### PR TITLE
JAMES-4054 Update SMIMECheckSignatureIntegrationTest to include IsSMIMESigned matcher

### PR DESCRIPTION
**Pull request by hungphan227**

### CODE REVIEW

File: mailet/crypto/src/main/java/org/apache/james/transport/matcher/IsSMIMESigned.java [hunk 1]
```diff
@@ -45,8 +45,7 @@ public Collection<MailAddress> match(Mail mail) throws MessagingException {
         if (message == null) {
             return null;
         }
-        
-        
+
```
Comment (chibenwa) [inline mailet/crypto/src/main/java/org/apache/james/transport/matcher/IsSMIMESigned.java:48]: Where are the (unit) tests for this class?

Comment (Arsnael) [reply mailet/crypto/src/main/java/org/apache/james/transport/matcher/IsSMIMESigned.java:48]: From what I see there is no unit tests in general at all for mailets and matchers declared in the crypto module, everything seems tested on an integration test level?

**Review by Arsnael — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

File: mailet/crypto/src/main/java/org/apache/james/transport/matcher/IsSMIMESigned.java (hunk 1 above)
Comment (chibenwa) [reply mailet/crypto/src/main/java/org/apache/james/transport/matcher/IsSMIMESigned.java:48]: THis one is a simple mailet only checking content type and its logic is not well exercised by the integration tests.

I do believe unit test makes full sense here and are quick and easy to write.

Comment (hungphan227) [reply mailet/crypto/src/main/java/org/apache/james/transport/matcher/IsSMIMESigned.java:48]: added

**Review by quantranhong1999 — APPROVED** [review]

File: mailet/crypto/src/test/java/org/apache/james/transport/matcher/IsSMIMESignedTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+public class IsSMIMESignedTest {
+    private IsSMIMESigned isSMIMESigned;
+
+    @BeforeEach
+    void beforeEach() {
+        isSMIMESigned = new IsSMIMESigned();
+    }
+
+    @Test
+    void matchShouldReturnNonEmptyListWhenMessageContentTypeIsMultipartSigned() throws Exception {
```
Comment (chibenwa) [inline mailet/crypto/src/test/java/org/apache/james/transport/matcher/IsSMIMESignedTest.java:40]: The same test, again and again...

How about using junit 5 parameters?

Comment (hungphan227) [reply mailet/crypto/src/test/java/org/apache/james/transport/matcher/IsSMIMESignedTest.java]: updated

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Pull request merged**